### DISCUSSION TITLE
增加更多可选的序列化方式

### DISCUSSION THREAD

**Discussion by SimTsai** [2881099/FreeRedis#34] [Ideas]
当前仅支持返回类型为string的序列化，是否可以考虑支持返回byte[]的序列化callback？

**Comment by 2881099** [comment]
之前为了考虑兼容 CSRedisCore 序列化，所以没有提供 byte[] 序列化参数，可以考虑。

**Reply by NMSAzulX** [reply]
> 之前为了考虑兼容 CSRedisCore 序列化，所以没有提供 byte[] 序列化参数，可以考虑。

建议支持 ReadOnlySpan<byte> / Span<byte> 参数的API ，后续协议解析后可能通过 Span<byte> 分片传入。

**Reply by 2881099** [reply]
这个看大后期

**Reply by NMSAzulX** [reply]
这个特性我已经拉到了Issue ，但暂未打上标记，我们需要对代码和需求进行追踪以方便后续的跟进和相似问题的管理，粗略的我们需要讨论实施优先级，对源代码结构的影响，以及UT测试。

**Comment by 2881099** [comment]
```c#
        public Func<object, object> Serialize;
        public Func<string, Type, object> Deserialize;
        public Func<byte[], Type, object> DeserializeRaw;
```

增加了一个新的反序列化设置，DeserializeRaw 针对  byte[] ，同时保留旧的 string 反序列化设置。

先讨论一波，如果可以再发布到 nuget 包。

**Reply by NMSAzulX** [reply]
现实情况可能比这个更加复杂，可能有额外配置。

**Reply by NMSAzulX** [reply]
如果使用委托来对接序列化方法。
严谨的考虑：这个委托恐怕不方便对外开放，我们需要考虑这个委托实现的极端情况对当前库的影响。
宽松的政策：放开且允许随意定制委托，我们不承担任何问题。

**Comment by 2881099** [comment]
请过目： https://github.com/2881099/FreeRedis/commit/add82f4846d0ba2211dc12dbf8c015069ff734c7

**Reply by NMSAzulX** [reply]
建议先引用到ISSUE，代码关联ISSUE，需求较少可以不定制里程碑，但发版需要提醒大家。 我们现在重建流程需要严谨一点，包括复杂的CI流程，这不会耽误太多时间。

**Comment by NMSAzulX** [comment]
@simhgd 题主是否有其他，更多，更具体，更大胆的想法。😁

**Comment by 2881099** [comment]
v0.2.1 先发布这个包了，使用 cli.DeserializeRaw 设置 byte[] 的反序列化。

**Comment by SimTsai** [comment]
是否可以文档中标明一种默认的 序列化及反序列化回调顺序
可以指定多个序列化回调方法， 按照优先级选择使用哪一种？
比如 [ReadOnly]Span -> byte[] -> string?
